src/leases/components/search/Search.js

```diff
@@ -105,7 +105,7 @@ class Search extends PureComponent<Props, State> {
       newState.municipalityOptions = getFieldOptions(props.leaseAttributes, LeaseFieldPaths.MUNICIPALITY);
       newState.tenantTypeOptions = getFieldOptions(props.leaseAttributes, LeaseTenantContactSetFieldPaths.TYPE, false);
       newState.typeOptions = getFieldOptions(props.leaseAttributes, LeaseFieldPaths.TYPE);
-      newState.serviceUnitOptions = getFieldOptions(props.leaseAttributes, 'service_unit', true);
+      //newState.serviceUnitOptions = getFieldOptions(props.leaseAttributes, 'service_unit', true);
```

Comment:
`isSearchBasicMode` method beginning on line 147 `delete searchQuery.service_unit;` needs to be added so that the detailed search view would not render when the service unit filter is being modified. Now it will render when the page is refreshed.

Reply:
fixed